Fix dynmap stack trace getting printed when dynmap isn't installed

[14:17:58 WARN]: [MedievalFactions] Task #5 for MedievalFactions v4.0-alpha-5 generated an exception
java.lang.NoClassDefFoundError: org/dynmap/markers/MarkerSet
        at dansplugins.factionsystem.DynmapManager.dynmapUpdateFaction(DynmapManager.java:341) ~[?:?]
        at dansplugins.factionsystem.DynmapManager.dynmapUpdateFactions(DynmapManager.java:150) ~[?:?]
        at dansplugins.factionsystem.DynmapManager.access$000(DynmapManager.java:18) ~[?:?]
        at dansplugins.factionsystem.DynmapManager$1.run(DynmapManager.java:55) ~[?:?]
        at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftTask.run(CraftTask.java:76) ~[spigot-1.12.2.jar:git-Spigot-79a30d7-acbc348]
        at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:361) [spigot-1.12.2.jar:git-Spigot-79a30d7-acbc348]
        at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:739) [spigot-1.12.2.jar:git-Spigot-79a30d7-acbc348]
        at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:406) [spigot-1.12.2.jar:git-Spigot-79a30d7-acbc348]
        at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:679) [spigot-1.12.2.jar:git-Spigot-79a30d7-acbc348]
        at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:577) [spigot-1.12.2.jar:git-Spigot-79a30d7-acbc348]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_251]
Caused by: java.lang.ClassNotFoundException: org.dynmap.markers.MarkerSet
        at java.net.URLClassLoader.findClass(Unknown Source) ~[?:1.8.0_251]
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:152) ~[spigot-1.12.2.jar:git-Spigot-79a30d7-acbc348]
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:100) ~[spigot-1.12.2.jar:git-Spigot-79a30d7-acbc348]
        at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_251]
        at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_251]
        ... 11 more

--- a/src/main/java/dansplugins/factionsystem/DynmapManager.java
+++ b/src/main/java/dansplugins/factionsystem/DynmapManager.java
@@ -47,10 +47,13 @@ public class DynmapManager {
      * to be rendered on dynmap.
      * @param interval Number of ticks before the scheduled task executes again.
      */
-    public static void scheduleClaimsUpdate(long interval) {
+    public void scheduleClaimsUpdate(long interval) {
         new BukkitRunnable() {
             @Override
             public void run() {
+                if (!isDynmapPresent()) {
+                    return;
+                }
                 if (DynmapManager.getInstance().updateClaimsAreaMarkers) {
                     DynmapManager.getInstance().dynmapUpdateFactions();
                     DynmapManager.getInstance().updateClaimsAreaMarkers = false;
@@ -63,7 +66,11 @@ public class DynmapManager {
      * Tell the scheduled task that we have made changes and it should update the
      * area markers.
      */
-    public static void updateClaims() {
+    public void updateClaims() {
+        if (!isDynmapPresent()) {
+            return;
+        }
+
         if (DynmapManager.hasDynmap()) {
             DynmapManager.getInstance().updateClaimsAreaMarkers = true;
         }
@@ -80,9 +87,11 @@ public class DynmapManager {
 
     public DynmapManager() {
         PluginManager pm = getServer().getPluginManager();
+
         /* Get dynmap */
         dynmap = pm.getPlugin("dynmap");
-        if(dynmap == null) {
+
+        if(!isDynmapPresent()) {
             System.out.println(LocaleManager.getInstance().getText("CannotFindDynmap"));
         }
         else {
@@ -105,6 +114,10 @@ public class DynmapManager {
                 System.out.println(LocaleManager.getInstance().getText("ErrorIntegratingWithDynmap") + e.getMessage());
             }
         }
+    }
+
+    private boolean isDynmapPresent() {
+        return (dynmap != null);
     }
 
     private MarkerAPI getMarkerAPI() {

--- a/src/main/java/dansplugins/factionsystem/MedievalFactions.java
+++ b/src/main/java/dansplugins/factionsystem/MedievalFactions.java
@@ -57,8 +57,8 @@ public class MedievalFactions extends JavaPlugin {
         Metrics metrics = new Metrics(this, pluginId);
 
         if (DynmapManager.hasDynmap()) {
-            DynmapManager.scheduleClaimsUpdate(600); // Check once every 30 seconds for updates.
-            DynmapManager.updateClaims();
+            DynmapManager.getInstance().scheduleClaimsUpdate(600); // Check once every 30 seconds for updates.
+            DynmapManager.getInstance().updateClaims();
         }
     }
 

--- a/src/main/java/dansplugins/factionsystem/Scheduler.java
+++ b/src/main/java/dansplugins/factionsystem/Scheduler.java
@@ -136,7 +136,7 @@ public class Scheduler {
         if (factionToRemove != null) {
             // remove claimed land objects associated with this faction
             ChunkManager.getInstance().removeAllClaimedChunks(factionToRemove.getName(), PersistentData.getInstance().getClaimedChunks());
-            DynmapManager.updateClaims();
+            DynmapManager.getInstance().updateClaims();
 
             // remove locks associated with this faction
             PersistentData.getInstance().removeAllLocks(factionToRemove.getName());

--- a/src/main/java/dansplugins/factionsystem/commands/ClaimCommand.java
+++ b/src/main/java/dansplugins/factionsystem/commands/ClaimCommand.java
@@ -52,7 +52,7 @@ public class ClaimCommand {
             ChunkManager.getInstance().radiusClaimAtLocation(depth, player, player.getLocation(), playersFaction);
         }
 
-        DynmapManager.updateClaims();
+        DynmapManager.getInstance().updateClaims();
         return true;
     }
 

--- a/src/main/java/dansplugins/factionsystem/commands/DisbandCommand.java
+++ b/src/main/java/dansplugins/factionsystem/commands/DisbandCommand.java
@@ -80,7 +80,7 @@ public class DisbandCommand {
 
         // remove claimed land objects associated with this faction
         ChunkManager.getInstance().removeAllClaimedChunks(nameOfFactionToRemove, PersistentData.getInstance().getClaimedChunks());
-        DynmapManager.updateClaims();
+        DynmapManager.getInstance().updateClaims();
 
         // remove locks associated with this faction
         PersistentData.getInstance().removeAllLocks(PersistentData.getInstance().getFactions().get(i).getName());

--- a/src/main/java/dansplugins/factionsystem/commands/LeaveCommand.java
+++ b/src/main/java/dansplugins/factionsystem/commands/LeaveCommand.java
@@ -51,7 +51,7 @@ public class LeaveCommand {
 
                                     // remove claimed land objects associated with this faction
                                     ChunkManager.getInstance().removeAllClaimedChunks(PersistentData.getInstance().getFactions().get(i).getName(), PersistentData.getInstance().getClaimedChunks());
-                                    DynmapManager.updateClaims();
+                                    DynmapManager.getInstance().updateClaims();
 
                                     PersistentData.getInstance().getFactions().get(i).removeMember(player.getUniqueId(), PersistentData.getInstance().getPlayersPowerRecord(player.getUniqueId()).getPowerLevel());
                                     PersistentData.getInstance().getFactions().remove(i);

--- a/src/main/java/dansplugins/factionsystem/commands/UnclaimCommand.java
+++ b/src/main/java/dansplugins/factionsystem/commands/UnclaimCommand.java
@@ -16,7 +16,7 @@ public class UnclaimCommand {
                 Player player = (Player) sender;
                 if (PersistentData.getInstance().isInFaction(player.getUniqueId())) {
                     ChunkManager.getInstance().removeChunkAtPlayerLocation(player);
-                    DynmapManager.updateClaims();
+                    DynmapManager.getInstance().updateClaims();
                     return true;
                 }
                 else {

--- a/src/main/java/dansplugins/factionsystem/commands/UnclaimallCommand.java
+++ b/src/main/java/dansplugins/factionsystem/commands/UnclaimallCommand.java
@@ -32,7 +32,7 @@ public class UnclaimallCommand {
 
                         // remove claimed chunks
                         ChunkManager.getInstance().removeAllClaimedChunks(faction.getName(), PersistentData.getInstance().getClaimedChunks());
-                        DynmapManager.updateClaims();
+                        DynmapManager.getInstance().updateClaims();
                         player.sendMessage(ChatColor.GREEN + String.format(LocaleManager.getInstance().getText("AllLandUnclaimedFrom"), factionName));
 
                         // remove locks associated with this faction
@@ -58,7 +58,7 @@ public class UnclaimallCommand {
 
                         // remove claimed chunks
                         ChunkManager.getInstance().removeAllClaimedChunks(faction.getName(), PersistentData.getInstance().getClaimedChunks());
-                        DynmapManager.updateClaims();
+                        DynmapManager.getInstance().updateClaims();
                         player.sendMessage(ChatColor.GREEN + LocaleManager.getInstance().getText("AllLandUnclaimed"));
 
                         // remove locks associated with this faction

--- a/src/main/java/dansplugins/factionsystem/eventhandlers/MoveHandler.java
+++ b/src/main/java/dansplugins/factionsystem/eventhandlers/MoveHandler.java
@@ -40,7 +40,7 @@ public class MoveHandler implements Listener {
                             public void run() {
                                 // add new chunk to claimed chunks
                                 ChunkManager.getInstance().claimChunkAtLocation(player, player.getLocation(), faction);
-                                DynmapManager.updateClaims();
+                                DynmapManager.getInstance().updateClaims();
                             }
                         }, 1); // delayed by 1 tick (1/20th of a second) because otherwise players will claim the chunk they just left
                     }

